iris_grib/_load_convert.py

```diff
@@ -843,24 +843,18 @@ def grid_definition_template_12(section, metadata):
     # Check whether Di and Dj are as consistent as possible with that
     # interpretation - i.e. they are within 1cm.
     def check_range(v1, v2, n, d):
-        min_last = v1 + (n - 1) * (d - 1)
-        max_last = v1 + (n - 1) * (d + 1)
-        if not (min_last < v2 < max_last):
+        small = min(v1, v2)
+        large = max(v1, v2)
```

Comment:
The only thing that is slightly bothering me is if someone created a file where, for example, 
x1=0, x2=10 but they set the scanning mode as -x
but I suppose that isn't the job of this function to catch anyway. In the above case we would end up creating a coord of (0->10) and ignore the scanning mode.

Although it does point out another over-specification of the spec: why do you need to know the scanning direction if you have the location of the first and last point?

Reply:
My concern with this idea is that I think the data in question do not have the scanning mode set to "negative", so would fail this check.  I would prefer that these checks occurred after the user-supplied `callback` function, so that it becomes possible to fix known meta-data errors, but this probably isn't practical. As an alternative, perhaps a scanning-mode mismatch could be a Warning instead of an Exception?

Reply:
FWIW my opinion: I think we already know that (a) the spec has redundancy and (b) files not fully conforming can be expected.  So it makes sense to do our best to make sense of what there is, but also emit a warning if not all information is consistent.

The existing code derives direction from Y1 + Y2, and spacing from "NY", and that works fine.  
It is not possible to have negative "DY" anyway, so Y1 and Y2 give a definitive scanning direction and you can't really get it wrong.  
I.E. if Y1 and Y2 were "intended" to be the other way round, this can only be indicated in the scanning direction.  But this would really violate the GDT description, which describes Y1 and Y2 as the "first" and "last" gridpoints.

So, I think we can safely just use Y1, Y2 and NY to define the coordinate.
I would then merely check that DY and scanning mode match the other evidence, and emit warnings if not.

Reply:
(all of which goes equally for the X grid, obviously)

Reply:
febc77d7b6691e5f806b214d737ccce378253330